tests/pytests/integration/minion/test_minion_module.py

```diff
@@ -0,0 +1,86 @@
+import shutil
+import time
+from pathlib import Path
+
+import pytest
+
+
+@pytest.fixture(scope="module")
+def master_minion(salt_master):
+    config_overrides = {
+        "root_dir": salt_master.config["root_dir"],
+        "return_retry_timer_max": 0,
+        "return_retry_timer": 5,
+        "return_retry_tries": 30,
+    }
+    factory = salt_master.salt_minion_daemon(
+        salt_master.id,
+        overrides=config_overrides,
+        extra_cli_arguments_after_first_start_failure=["--log-level=info"],
+    )
+    factory.after_terminate(
+        pytest.helpers.remove_stale_minion_key, salt_master, factory.id
+    )
+
+    with factory.started():
+        yield factory
+
+
+@pytest.fixture(scope="module")
+def second_minion(salt_master, salt_minion_id):
+    config_overrides = {
+        "return_retry_timer_max": 0,
+        "return_retry_timer": 5,
+        "return_retry_tries": 30,
+    }
+    factory = salt_master.salt_minion_daemon(
+        salt_minion_id,
```

Comment:
```suggestion
        salt_minion_id,
```
During integration tests, there's a a salt master and a salt minion running, `salt_minion_id` is whats used for the minion ID.

Why are we reusing that `salt_minion_id` as the second master here?

Reply:
If you look in the `master_minion` function you'll see that I'm using the master ID as the ID for the minion that's started up with it, which allows them to share the same directory in the test suite. That means the generated `salt_minion_id` is free to use for the second minion.

Reply:
That is only true if no other tests run.
If other tests run, for example, before this one, the minion with the ID of the value of `salt_minion_id` would already be running, and reusing that ID would mean two minions with the same ID(Don't really know how salt handles that)